book/content/material/material_advanced.md

```diff
@@ -0,0 +1,273 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.17.3
+kernelspec:
+  display_name: festim-workshop
+  language: python
+  name: python3
+---
+
+# Advanced Material Features #
+
++++
+
+This tutorial introduces more advanced ways of handling materials in FESTIM. A key addition in FESTIM 2.0 is the ability to perform multi-species simulations, which often requires assigning distinct material properties to each species (see __[documentation](https://festim.readthedocs.io/en/v2.0-alpha.3/api/index.html#festim.material.Material)__ for more information). In some cases, users may also need to define custom material properties—for example, specifying a turbulence-dependent viscosity or user-defined diffusivity.
+
+Objectives:
+* Assign different material properties to different species
+* Solve a problem with custom material properties (e.g., diffusivity)
+
++++
+
+## Assigning different material properties in a multi-species simulation ##
+
+Consider the following 1D example that simulates the diffusion of protium, deuterium and tritium species:
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+import festim as F
+import numpy as np
+
+my_model = F.HydrogenTransportProblem()
+
+protium = F.Species("H")
+deuterium = F.Species("D")
+tritium = F.Species("T")
+my_model.species = [protium, deuterium, tritium]
+
+my_model.mesh = F.Mesh1D(np.linspace(0, 1, 100))
+
+left_surf = F.SurfaceSubdomain1D(id=1, x=0)
+right_surf = F.SurfaceSubdomain1D(id=2, x=1)
+```
+
+Here we define the different diffusivities for each species. The `mat` dictionary stores keys for each species' diffusion coefficient `D_0` and activation energy `E_D`. We define one domain for all species called `bulk` that serves as the volume for the material:
```

Comment:
```suggestion
Here we define the different diffusivities (or rather the diffusivities pre-exponential factors) for each species.
Instead of passing a single value to the `D_0` of `Material`, we pass a dictionary where the keys are the different `Species` objects and values are values of pre-exponential factors `D_0` and activation energy `E_D`.
We define one domain called `bulk` and attach the material to it:
```